demo-client/src/main/java/com/illia/client/service/FileHandlingService.java

```diff
@@ -2,47 +2,65 @@
 
 import com.illia.client.config.ClientConfig;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.multipart.MultipartFile;
 
 import java.io.ByteArrayInputStream;
-import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.stream.Stream;
 
 @Service
 public class FileHandlingService {
 
     @Autowired
-    ClientConfig clientConfig;
-
-    // needed?
-    public File resolveMultipartFile(String fileName, MultipartFile multipartFile) throws IOException {
-
-        File file = null;
-        if (multipartFile != null) {
-            file = new File(fileName);
-            try (var is = multipartFile.getInputStream();
-                 var os = new FileOutputStream(file)) {
-                is.transferTo(os);
-                os.flush();
-            }
-        }
-        return file;
+    private ClientConfig clientConfig;
+
+    public ByteArrayResource resolveMultipartFile(MultipartFile multipartFile) throws IOException {
+        return new ByteArrayResource(multipartFile.getBytes());
     }
 
-    public boolean saveFile(String fileName, byte[] content) throws IOException{
-        var filePath = Path.of(clientConfig.getDownloadedFilesDirectoryPrefix(), fileName);
-        var fileAlreadyExists = Files.deleteIfExists(filePath);
+    public boolean saveFile(String fileName, byte[] content, boolean overwrite) throws IOException {
+        var filePath = resolvePath(fileName);
+
+        if (overwrite) {
+            Files.deleteIfExists(filePath);
+        }
 
-        try(var os = new FileOutputStream(filePath.toFile());
-            var is = new ByteArrayInputStream(content)){
+        try (var os = new FileOutputStream(filePath.toFile());
+             var is = new ByteArrayInputStream(content)) {
             is.transferTo(os);
             os.flush();
+            return true;
+        }catch (Exception e){
+            return false;
+        }
+    }
+
+    public boolean deleteFile(String fileName) throws IOException {
+        return Files.deleteIfExists(resolvePath(fileName));
+    }
+
+    public Path resolveFilePath(String fileName) {
+        if (fileName != null && Files.exists(resolvePath(fileName))) {
+            return resolvePath(fileName);
         }
-        return !fileAlreadyExists;
+        return null;
     }
 
+    public boolean exists(String fileName) {
+        return fileName != null && Files.exists(resolvePath(fileName));
+    }
+
+    private Path resolvePath(String... args) {
```

Comment:
path string could be produced by simple:
clientConfig.getDownloadedFilesDirectoryPrefix() + "/" + String.join("/", args);